lib/GLPI/Agent/SNMP/MibSupport/BrotherNetConfig.pm

```diff
@@ -77,6 +81,34 @@ sub getModel {
     return $model;
 }
 
+sub getPortIfType {
+    my ($self) = @_;
+
+    my $device = $self->device
+        or return;
+
+    # Get list of device ports
+    my $ports = $device->{PORTS}->{PORT};
+
+    # Get list of device ports types (lan(1)/wirelesslan(2))
+    my $brMultiIFType = $self->walk(brMultiIFType) || {};
+
+    # Get list of device ports names
+    my $brMultiIFNodeType = $self->walk(brMultiIFNodeType) || {};
+
+    foreach my $index (keys %{$brMultiIFType}) {
+        foreach my $port (keys %{$ports}) {
+            if ($ports->{$port}->{IFNAME} eq $brMultiIFNodeType->{$index}) {
```

Comment:
Here you have no guarantee hash fields are defined and comparing not defined values will fail with an error. So you just have to check they are defined.
```suggestion
        foreach my $port (keys %{$ports}) {
            next unless defined($ports->{$port}->{IFNAME}) && defined($brMultiIFNodeType->{$index});
            if ($ports->{$port}->{IFNAME} eq $brMultiIFNodeType->{$index}) {
```